### PR TITLE
Remove "Shared Environments" concept from new backend system

### PR DESCRIPTION
**Pull request by tudi2d**
We have decided to remove shared environments as we are exploring new concepts for achieving cross-deployment configuration realated to https://github.com/backstage/backstage/issues/18372

This includes removing `createSharedEnvironment` and any related types, as well as the `env` option to `createBackend`.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend-defaults/src/CreateBackend.test.ts [hunk 1]
```diff
@@ -86,65 +85,34 @@ describe('createBackend', () => {
...
-      "Shared environment version '[object Object]' is invalid or not supported",
-    );
-    expect(() =>
-      createBackend({
-        env: { version: 'v2' } as any,
-      }),
-    ).toThrow("Shared environment version 'v2' is invalid or not supported");
-  });
-
   it('should prioritize services correctly', async () => {
```
Comment (tudi2d) [inline packages/backend-defaults/src/CreateBackend.test.ts:88]: Unsure if this test is still relevant/logicaly needed here

Comment (Rugvip) [reply packages/backend-defaults/src/CreateBackend.test.ts:88]: Yep I think we can remove it completely at this point

**Review by Rugvip — COMMENTED** [review]
Nice! 👍

File: .changeset/rich-zoos-occur.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/backend-plugin-api': minor
+'@backstage/backend-defaults': minor
+---
+
+Removing shared environments concept from the new experimental backend system.
```
Comment (freben) [inline .changeset/rich-zoos-occur.md:6]: This should be prefixed with `**BREAKING**:` and then let's highlight it in the release notes eventually in the mainline release.

**Review by freben — COMMENTED** [review]
Cool! Just a changeset tweak needed.

**Review by freben — APPROVED** [review]
⭐

**Pull request merged**